tests/cluster/tests/24-links.tcl

```diff
@@ -0,0 +1,106 @@
+# Check cluster info stats
+
+source "../tests/includes/init-tests.tcl"
+
+test "Create a cluster with two single-node shards" {
+    create_cluster 2 0
+}
+
+test "Cluster should start ok" {
+    assert_cluster_state ok
+}
+
+test "Each node has two links with each peer" {
+    foreach_redis_id id {
+        # Get number of peers, excluding myself
+        set nodes [get_cluster_nodes $id]
+        set num_peers [expr [llength $nodes] - 1]
+        
+        # Get number of links to peers
+        set links [get_cluster_links $id]
+        set num_links [llength $links]
+
+        # Two links per peer
+        assert {$num_peers*2 eq $num_links}
+
+        # For each peer there should be exactly one
+        # link "to" it and one link "from" it.
+        foreach n $nodes {
+            if {[has_flag $n myself]} continue
+            set peer [dict get $n id]
+            set to 0
+            set from 0
+            foreach l $links {
+                if {[dict get $l node] eq $peer} {
+                    if {[dict get $l dir] eq "to"} {
+                        incr to
+                    } elseif {[dict get $l dir] eq "from"} {
+                        incr from
+                    }
+                }
+            }
+            assert {$to eq 1}
+            assert {$from eq 1}
+        }
+    }
+}
+
+set primary1_id 0
+set primary2_id 1
+
+set primary1 [Rn $primary1_id]
+set primary2 [Rn $primary2_id]
+
+test "Disconnect link when send buffer limit reached" {
+    # Set timeout to 1 hour so links won't get disconnected due to timeouts
+    set_cluster_node_timeout [expr 60*60*1000]
```

Comment:
Redis cluster tests are a little dumb in that they don't reset between tests, so you need to reset this manually. Same with the sendbuf later.

Reply:
Good point. Done.